test/factbase/test_lazy_taped.rb

```diff
@@ -0,0 +1,252 @@
+# frozen_string_literal: true
+
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 Yegor Bugayenko
+# SPDX-License-Identifier: MIT
+
+require_relative '../test__helper'
+require_relative '../../lib/factbase'
+require_relative '../../lib/factbase/lazy_taped'
+
+# Test.
+# Author:: Yegor Bugayenko (yegor256@gmail.com)
+# Copyright:: Copyright (c) 2024-2025 Yegor Bugayenko
+# License:: MIT
+class TestLazyTaped < Factbase::Test
+  def test_tracks_insertion
+    t = Factbase::LazyTaped.new([])
+    t << {}
+    assert_equal(1, t.inserted.size)
+  end
+
+  def test_does_not_copy_on_read
+    original = [{ foo: [1, 2] }]
+    t = Factbase::LazyTaped.new(original)
+    assert_equal(1, t.size)
+    refute_empty(t)
+    t.each { |m| m['foo'] }
+    assert_empty(t.pairs)
+  end
+
+  def test_copies_on_insert
+    original = [{ foo: [1] }]
+    t = Factbase::LazyTaped.new(original)
+    t << { bar: [2] }
+    refute_empty(t.pairs)
+    assert_equal(2, t.size)
+  end
+
+  def test_copies_on_property_set
+    original = [{ 'foo' => [1] }]
+    t = Factbase::LazyTaped.new(original)
+    t.each { |m| m['bar'] = 42 }
+    refute_empty(t.pairs)
+    assert_equal(1, t.added.size)
+    # Original should not be modified
```

Comment:
@HaidarJbeily7 could you pass this text as an assertion message instead of using comments?

```ruby
assert_nil(original[0]['bar'], "Original should not be modified")
```